src/python/pants/engine/internals/target_adaptor.py

```diff
@@ -104,10 +104,19 @@ def __init__(
     ) -> None:
         self.type_alias = type_alias
         self.name = name
-        self.kwargs = kwargs
+        self.kwargs = FrozenDict.deep_freeze(kwargs)
         self.description_of_origin = __description_of_origin__
         self.origin_sources_blocks = __origin_sources_blocks__
 
+    def with_new_kwargs(self, **kwargs) -> TargetAdaptor:
+        return TargetAdaptor(
+            type_alias=self.type_alias,
+            name=self.name,
+            kwargs=kwargs,
```

Comment:
Should `FrozenDict.deep_freeze` be applied here to?

Reply:
Yes, good catch. I mean, in practice it's OK because we only use this on kwargs that were already frozen, but it's a good idea to do so anyway.

Reply:
Oh, I guess the ctor applies deep_freeze for us, so no need to do so here.